core/src/consensus.rs

```diff
@@ -520,9 +530,16 @@ impl Tower {
                             // all of them.
                             panic!("no ancestors found with slot: {}", last_voted_slot);
                         } else {
-                            // bank_forks doesn't have corresponding data for the stray restored last vote,
-                            // meaning some inconsistency between saved tower and ledger.
-                            // (newer snapshot, or only a saved tower is moved over to new setup?)
+                            // compare slots not to error! just because of newer snapshots
+                            if self.is_first_switch_check() && switch_slot < last_voted_slot {
```

Comment:
hmmm will `switch_slot < last_voted_slot` trigger if there's a heavier fork at a lesser slot? I.e. something that looks like:

```
                                       5 (last vote)
                                    /
0 - 1 (snapshot root)
                                  \
                                      2 (new heaviest fork)
```


This might happen because `check_switch_threshold` is called here: https://github.com/solana-labs/solana/blob/master/core/src/replay_stage.rs#L1545-L1553 even if we are locked out detected here: https://github.com/solana-labs/solana/blob/master/core/src/replay_stage.rs#L1578

Reply:
@carllin yeah that condition will trigger. Even under that condition, I think we should do `error!` a log (note: not `panic!`) because it's still a bad condition where there is no data in bank_forks for switch_slot (this is in the fallback closure for unwrap_or_else). So, I'm saying that this case isn't a false positive. What I wanted to avoid is a different false positive where we're resuming from snapshot (=~ `switch_slot`) which is newer than  or equal to `last_vote`.

Also, fyi, this `else` clause is entered only if we haven't any single vote since restart and make_check_switch_threshold_decision is called for the first time.

Reply:
oh, you're saying that we should avoid `error!` here if we're locked out under that situation because there is no possibility of bad voting? So, are you thinking this is kind of a false positive in fact?

Well, regarding for that thinking, that's the reason I made this `error!` not `panic!`.

Reply:
@ryoqun yeah, was mostly just concerned about the false positive.

How about lowering to a `warn` and adding a comment?

Reply:
@carllin done accordingly: b51b9daf67. How does that look? I also added `datapoint_warn` to monitor for fun. :)